# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/linux-header/ps3fb-ioctls.patch
# Copyright (C) 2020 - 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./arch/powerpc/include/uapi/asm/ps3fb.h	2012-07-21 22:58:29.000000000 +0200
+++ ./arch/powerpc/include/uapi/asm/ps3fb.h	2018-01-27 14:10:55.546169979 +0100
@@ -29,6 +29,9 @@
 #define PS3FB_IOCTL_ON            _IO('r', 4)        /* use IOCTL_FSEL */
 #define PS3FB_IOCTL_OFF           _IO('r', 5)        /* return to normal-flip */
 #define PS3FB_IOCTL_FSEL          _IOW('r', 6, int)  /* blit and flip request */
+#define PS3FB_IOCTL_GPU_SETUP     _IO('r', 7)         /* enable FIFO access */
+#define PS3FB_IOCTL_GPU_INFO      _IOR('r', 8, int)   /* get GPU info */
+#define PS3FB_IOCTL_GPU_ATTR      _IOW('r', 9, int)   /* set attribute */
 
 #ifndef FBIO_WAITFORVSYNC
 #define FBIO_WAITFORVSYNC         _IOW('F', 0x20, __u32) /* wait for vsync */
@@ -42,4 +45,22 @@
 	__u32 num_frames; /* num of frame buffers */
 };
 
+struct ps3fb_ioctl_gpu_info {
+	__u32 vram_size;      /* size of available video memory */
+	__u32 fifo_size;      /* size of command buffer */
+	__u32 ctrl_size;      /* size of dma control registers */
+	__u32 dinfo_size;     /* size of driver info */
+	__u32 reports_size;   /* size of reports */
+	__u32 device_size[8]; /* size of gpu devices */
+};
+
+struct ps3fb_ioctl_gpu_attr {
+	__u64 attr;           /* attribute */
+	__u64 p0;             /* 1st parameter */
+	__u64 p1;             /* 2nd parameter */
+	__u64 p2;             /* 3rd parameter */
+	__u64 p3;             /* 4th parameter */
+};
+
+
 #endif /* _ASM_POWERPC_PS3FB_H_ */
 
